lib/rubocop/cop/lint/literal_in_interpolation.rb

```diff
@@ -91,7 +93,16 @@ def autocorrected_value_for_symbol(node)
           end_pos =
             node.loc.end ? node.loc.end.begin_pos : node.loc.expression.end_pos
 
-          range_between(node.loc.begin.end_pos, end_pos).source
+          range_between(node.loc.begin.end_pos, end_pos).source.gsub('"', '\"')
+        end
+
+        def autocorrected_value_in_hash_for_symbol(node)
+          # TODO: We need to detect symbol unacceptable names more correctly
+          if / |"|'/.match?(node.value.to_s)
```

Comment:
A gem [unparser](https://github.com/mbj/unparser) provides more simple (and better) way to generate code strings for autocorrect result. 
Should we add it to the list of dependencies?
```suggestion
            Unparser.unparse(node).gsub('"') { '\"' }
```

Reply:
Probably that'd be an overkill for a single usage. I'm always wary of adding additional runtime deps.

Reply:
I aimed to separate codes to unparse hash from the rubocop to make its project scope more clear.
But after more research I realized that the `unparser gem` sometimes generate results not exact same to Ruby's one.
Thus I agree to cancel this suggestion.

Reply:
@KessaPassa Unparser only guarantess: `parse(code) == parse(unparse(parse(code)))`, so it guarantees you generate concrete syntax that if parsed produces the original AST, not the same concrete syntax. If you knew the concrete syntax already there is no need to unparse in the first place ;)